src/IndexingMsg.chpl

```diff
@@ -1328,31 +980,31 @@ module IndexingMsg
                 return ivBoolHelper(bool);
             }
             when (DType.BigInt, DType.Int64, DType.BigInt) {
-              return bigintIvInt64Helper(bigint);
+                return ivInt64Helper(bigint);
             }
             when (DType.BigInt, DType.Int64, DType.Int64) {
-              return bigintIvInt64Helper(int);
+                return ivInt64Helper(int);
             }
             when (DType.BigInt, DType.Int64, DType.UInt64) {
-              return bigintIvInt64Helper(uint);
+                return ivInt64Helper(uint);
             }
```

Comment:
there's an issue here. I don't know why the compiler doesn't complain, but we have the same aggregation into a bigint array from a non-bigint type. I just pulled down this PR and verified that the below block of code causes the server to hang

```python
>>> a = ak.arange(2**80, 2**80+ 10)
>>> a[ak.arange(5,10)] = ak.arange(5,10)
```
On master this gives us the expected result
```python
>>> a = ak.arange(2**80, 2**80+ 10)
>>> a[ak.arange(5,10)] = ak.arange(5,10)
>>> a
array(["1208925819614629174706176" "1208925819614629174706177" "1208925819614629174706178" "1208925819614629174706179" "1208925819614629174706180" "5" "6" "7" "8" "9"])
```


We prob want to do something similar to `setPdarrayIndexToValueMsg` where we have a dtype const and set it to something like
```chapel
const dtype = if gX.dtype == DType.BigInt then DType.BigInt else gY.dtype;
```
then we'll need to update the select statement to be `select(gX.dtype, gIV.dtype, dtype)`. And change the helper functions to cast the type to dtype in the `agg.copy`

Once we do that we can remove the cases that aren't `when (DType.BigInt, indexType, DType.BigInt)`

Reply:
Good catch @pierce314159.